File: python/lsearch.py
#!/usr/bin/env python3
import random

random.seed(42)

# linear search

def lsearch(A: list[int], target: int) -> int:
	size = len(A)
	for i in range(size):
		if target == A[i]:
			print("Element found at index", i)
			return i
	print("Element not found.")

def main():
	N = 100

	A = sorted([random.randint(1, N + N) for _ in range(N)])
	target = A[-1] # last element in list A (worst case)

	lsearch(A, target)



"""
Time Complexity:

	best case: O(1)
	worst case: O(N)
	average case: O(N)

"""

if __name__ == "__main__":
	main()

File: c/lsearch.c
#include <stdio.h>

// Linear Search

int main(){
	int N[7] = {1, 3, 5, 7, 13, 15, 18};
	int target = 18; 

	int len = sizeof(N) / sizeof(N[0]);

	for(int i = 0; i < len; i++){
		if(target == N[i]){
			printf("Element Found at %d \n", i);
			return 0;
		}
		
	}
	printf("Element not found.");

}


/* 

Time Complexity :

	-> Best case: O(1)
	-> Worst case: O(n) 

*/